RELEASES.md

```diff
@@ -1,4 +1,48 @@
-# Release rayon 1.0.3
+# Release rayon 1.1.0 / rayon-core 1.5.0 (2019-06-05)
```

Comment:
I'll check this date before we actually merge and publish.

Reply:
```suggestion
# Release rayon 1.1.0 / rayon-core 1.5.0 (2019-06-12)
```